### PR TITLE
Purpose readme contribution

### PR DESCRIPTION
**Pull request by olegrjumin**
To make Underscore.js more approachable to a larger number of users—from informed developers to curious passersby and new and interested visitors—I have added a new section to the Underscore.js README. The section is entitled What to Use Underscore.js for and When to Use It. As the name suggests, the new section, a brief paragraph, explains in plain and straightforward language what Underscore.js is used for and when one should use it.

Other repositories may begin to use this new informative and catchy section soon, as my colleagues contact open-source technologies in an effort to make them more accessible to the tens of thousands of new users learning to code and thousands joining GitHub every day.

### CODE REVIEW

**Comment by jdalton** [comment]
Hi @olegrjumin & @jmatembu!

Shout out to https://learn.modern-developer.com/ :loudspeaker:

**Comment by jdalton** [comment]
Related to #2437 #2438 https://github.com/lodash/lodash/pull/1954.

**Comment by Dorian** [comment]
@kmoe :+1: I don't see the need to reference jQuery.

File: README.md [hunk 1]
```diff
@@ -12,6 +12,16 @@ Underscore.js is a utility-belt library for JavaScript that provides
 support for the usual functional suspects (each, map, reduce, filter...)
 without extending any core JavaScript objects.
 
+Underscore.js contains functions to perform common operations on data and it works on arrays, JSON objects, etc. It pairs nicely with jQuery and developers tend to use Underscore.js to manipulate the data first and then jQuery for laying out the results on a page
+
+### What to use Underscore.js for and when to use It
+
+* When you need to stick to vanilla JavaScript implementations of utility functions, not extending any core objects.
+* Since not all new JavaScript functionality is available in all browsers, Underscore.js will provide JavaScript funcionality that works across all browsers.
```
Comment (jfmengels) [inline README.md]: typo here: funcionality --> functionality

**Comment by Bloomca** [comment]
I definitely agree about de-emphasising about jQuery. It's too eye-catchy, so better to replace some agnostic _DOM-manipulate library_ or something else.

**Comment by jmatembu** [comment]
Thank you for your feedback. It's highly appreciated. Will be incorporating your suggestions soon.

**Comment by olegrjumin** [comment]
Thank you all for suggestions, we've made changes!

**Comment by akre54** [comment]
The best place for this stuff is on http://underscorejs.org (built from the index.html file in the repository).

It currently says:

> Underscore is a JavaScript library that provides a whole mess of useful functional programming helpers without extending any built-in objects. It’s the answer to the question: “If I sit down in front of a blank HTML page, and want to start being productive immediately, what do I need?” … and the tie to go along with jQuery's tux and Backbone's suspenders.

> Underscore provides over 100 functions that support both your favorite workaday functional helpers: map, filter, invoke — as well as more specialized goodies: function binding, javascript templating, creating quick indexes, deep equality testing, and so on.

If there's something you think is missing, we can make changes, but this doesn't seem to be an improvement to my eyes.

**Comment by akre54** [comment]
Side note: as @jdalton pointed out, we seem to have a few of these types of pull requests lately on Backbone and Underscore.

Is this a project for something or everyone was simultaneously excited to make changes to the docs?

**Comment by michaelficarra** [comment]
Agreed with @akre54.

**Pull request closed without merging**